src/components/MembershipStatusPopup.tsx

```
@@ -0,0 +1,194 @@
+      <div
+        className={`relative z-10 mx-4 w-full max-w-md overflow-hidden rounded-2xl bg-white shadow-2xl transition-all duration-200 ${visible ? 'scale-100 opacity-100' : 'scale-95 opacity-0'}`}
+      >
+        {/* Header */}
+        <div
+          className={`px-8 pt-8 pb-6 ${isPaid ? 'bg-gradient-to-br from-green-500 to-emerald-600' : 'bg-gradient-to-br from-gray-400 to-gray-500'}`}
+        >
+          <button
+            onClick={internalOnClose}
+            className="absolute top-3 right-3 rounded-lg p-1.5 text-white/70 transition-colors hover:bg-white/20 hover:text-white"
+            aria-label="Close"
+          >
```

Biome (2.3.13)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)